--- a/Source/WTF/wtf/CurrentTime.cpp
+++ b/Source/WTF/wtf/CurrentTime.cpp
@@ -118,7 +118,7 @@
         if (tickCount >= tickCountLast)
             tickCountElapsed = (tickCount - tickCountLast);
         else {
-            __int64 tickCountLarge = tickCount + 0x100000000I64;
+            __int64 tickCountLarge = tickCount + 0x100000000LL;
             tickCountElapsed = tickCountLarge - tickCountLast;
         }
 
